velox/common/memory/MemoryUsageTracker.h

```diff
@@ -158,26 +158,33 @@ class MemoryUsageTracker
     {
       std::lock_guard<std::mutex> l(mutex_);
       increment = reserveLocked(size);
-      minReservation_ += increment;
+      minReservation_ = reservation_.load();
     }
     if (increment) {
       checkAndPropagateReservationIncrement(increment, true);
     }
   }
 
-  // Release unused reservation. Used reservation will be released as the
-  // allocations are freed.
+  // If a minimum reservation has been set with reserve(), resets the
+  // minimum reservation. If the current usage is below the minimum
+  // reservation, decreases reservation and usage down to the rounded
+  // actual usage.
   void release() {
-    int64_t remaining;
+    int64_t freeable;
     {
+      if (!minReservation_) {
```

Comment:
This check can be moved to the very beginning of the method. It doesn't require a lock and we don't need to define freeable variable until after this check.